File: Starter Projects/Python/ConsequencesStarterClient/src/SocketClient.py
import json
import queue
from websocket import WebSocketApp, create_connection

from Messages import InboundMessage

try:
    import thread
except ImportError:
    import _thread as thread


class SocketClient:
    def __init__(self, uri):
        self.uri = uri
        self.messages = queue.Queue()
        self.ws = create_connection(self.uri)

    def receive(self) -> InboundMessage:
        json_message = self.ws.recv()
        json_obj = json.loads(json_message)
        message_obj = InboundMessage(message=json_obj["Message"])
        return message_obj

    def send(self, message):
        json_message = json.dumps(message)
        self.ws.send(json_message)

File: Starter Projects/Java/ConsequencesStarterClient/src/main/java/SocketClient.java
import com.google.gson.Gson;

import java.net.URI;
import java.net.http.HttpClient;
import java.net.http.WebSocket;
import java.util.concurrent.BlockingQueue;
import java.util.concurrent.CompletionStage;
import java.util.concurrent.LinkedBlockingDeque;

public class SocketClient {
    WebSocket ws;
    WebSocketClient wsc;
    Gson gson = new Gson();

    public SocketClient(String uri){
        wsc = new WebSocketClient();
        ws = HttpClient
                .newHttpClient()
                .newWebSocketBuilder()
                .buildAsync(URI.create(uri), wsc)
                .join();
    }

    public void Send(OutboundMessage message)
    {
        String jsonMessage = gson.toJson(message);
        ws.sendText(jsonMessage, true);
    }

    public InboundMessage Receive()
    {
        String jsonMessage = wsc.GetMessageFromQueue();
        return gson.fromJson(jsonMessage, InboundMessage.class);
    }

    private static class WebSocketClient implements WebSocket.Listener {
        private final BlockingQueue<String> messageList;

        public WebSocketClient()
        {
            messageList = new LinkedBlockingDeque<String>();
        }

        public String GetMessageFromQueue()
        {
            try {
                return messageList.take();
            }
            catch (InterruptedException e)
            {
                return "Failed to take message from queue";
            }
        }

        @Override
        public void onOpen(WebSocket webSocket) {
            System.out.println("Websocket open");
            WebSocket.Listener.super.onOpen(webSocket);
        }

        @Override
        public CompletionStage<?> onText(WebSocket webSocket, CharSequence data, boolean last) {
            messageList.add(data.toString());
            return WebSocket.Listener.super.onText(webSocket, data, last);
        }

        @Override
        public void onError(WebSocket webSocket, Throwable error) {
            System.out.println("Error: " + error.getMessage());
            WebSocket.Listener.super.onError(webSocket, error);
        }
    }
}
